src/atomate2/abinit/jobs/base.py

```diff
@@ -0,0 +1,349 @@
+        )
+        return maker
+
+    def __post_init__(self):
+        """Process post-init configuration."""
+        self.critical_events = [
+            as_event_class(ce_name) for ce_name in self.CRITICAL_EVENTS
+        ]
+        if self.input_set_generator is None:
+            raise InitializationError("Input set generator is not set.")
+
+    @property
+    def calc_type(self):
+        return self.input_set_generator.calc_type
+
+    @job
+    def make(
+        self,
+        structure: Optional[Structure] = None,
+        prev_outputs: Optional[Any] = None,
+        restart_from: Optional[Union[str]] = None,
+        history: Optional[JobHistory] = None,
+    ) -> Union[jobflow.Flow, jobflow.Job]:
+        """
+        Return an ABINIT jobflow.Job.
+
+        Parameters
+        ----------
+        structure : Structure
+            A pymatgen structure object.
+        prev_outputs : TODO: add description from sets.base
+        restart_from : TODO: add description from sets.base
+        history : JobHistory
+            A JobHistory object containing the history of this job.
+        """
+        # Setup job and get general job configuration
+        config = setup_job(
+            structure=structure,
+            prev_outputs=prev_outputs,
+            restart_from=restart_from,
+            history=history,
+            wall_time=self.wall_time,
+        )
+
+        # Write abinit input set
+        write_abinit_input_set(
+            structure=structure,
+            input_set_generator=self.input_set_generator,
+            prev_outputs=prev_outputs,
+            restart_from=restart_from,
+            directory=config.workdir,
+        )
+
+        # Run abinit
+        run_status = run_abinit(
+            abinit_cmd=SETTINGS.ABINIT_CMD,
+            mpirun_cmd=SETTINGS.ABINIT_MPIRUN_CMD,
+            wall_time=config.wall_time,
+            start_time=config.start_time,
+        )
+
+        # parse Abinit outputs
+        run_number = config.history.run_number
+        task_doc = AbinitTaskDocument.from_directory(
+            config.workdir,
+            critical_events=self.critical_events,
+            run_number=run_number,
+            run_status=run_status,
+        )
+        task_doc.task_label = self.name
+
+        dumpfn(jsanitize(task_doc.dict()), fn="task_document.json", indent=2)
+
+        return self.get_response(
+            task_document=task_doc,
+            history=config.history,
+            max_restarts=SETTINGS.ABINIT_MAX_RESTARTS,
+            prev_outputs=prev_outputs,
+        )
+
+    def get_response(
+        self,
+        task_document: AbinitTaskDocument,
+        history: JobHistory,
+        max_restarts: int = 5,
+        prev_outputs: Optional[List[str]] = None,
+    ):
+        """Get new job to restart abinit calculation."""
+        if task_document.state == Status.SUCCESS:
+            return Response(
+                output=task_document,
+                detour=None,
```

Comment:
You don't need to define the rest of these values because they are the defaults. Ditto for many of the other responses!